src/extension.ts

```diff
@@ -523,6 +523,37 @@ export function activate(context: vscode.ExtensionContext) {
     context.subscriptions.push(vscode.commands.registerCommand(commands.CommandsConsts.globalImportSnippets,
         async _ => handleCommand(() => commands.importSnippets(snippetsProvider))
     ));
+
+    context.subscriptions.push(vscode.languages.registerDocumentDropEditProvider({ language: '*', scheme: '*'}, new class DropEditorProvider implements vscode.DocumentDropEditProvider {
+       async provideDocumentDropEdits(
+          document: vscode.TextDocument,
+          position: vscode.Position,
+          dataTransfer: vscode.DataTransfer,
+          token: vscode.CancellationToken
+        ): Promise<vscode.DocumentDropEdit | undefined> {
+            const dataItem = dataTransfer.get('text/uri-list');
+            if (!dataItem) {
+                return;
+            }
+            try {
+                const text = await dataItem.asString();
+                const parsedSource = JSON.parse(text) as readonly Snippet[];
+                if (parsedSource?.length !== 1) {
+                    return;
+                }
+                const draggedSnippet = parsedSource[0];
+                if (draggedSnippet.folder) {
+                    return;
+                }
+                return {
+                    insertText: draggedSnippet.value ?? '',
```

Comment:
I see a different return expression in the documentation here https://github.com/microsoft/vscode-extension-samples/blob/main/drop-on-document/src/extension.ts#L35, I'm not sure what's the best option. Let's use the best one.

Reply:
I noticed that there are several snippet syntaxes available.

If we use draggedSnippet.value as insertText directly, we cannot take advantage of the snippets variable. If we use DocumentDropEdit to appendText or appendVariable, we need to know the type of the snippet, which can be difficult. Therefore, I use the same logic as the open command to insert the snippet by using vscode.commands.executeCommand("editor.action.insertSnippet", xx) or vscode.window.activeTextEditor.edit(edit => { edit.insert(position, xxx); }).